packages/gui/src/components/nfts/detail/NFTDetailV2.tsx

```diff
@@ -21,34 +30,115 @@ import useNFTMetadata from '../../../hooks/useNFTMetadata';
 import NFTContextualActions, {
   NFTContextualActionTypes,
 } from '../NFTContextualActions';
-import NFTPreviewDialog from '../NFTPreviewDialog';
+import NFTProgressBar from '../NFTProgressBar';
+
+const ipcRenderer = (window as any).ipcRenderer;
 
 export default function NFTDetail() {
   const { nftId } = useParams();
   const { wallets: nftWallets, isLoading: isLoadingWallets } =
     useGetNFTWallets();
-  const openDialog = useOpenDialog();
   const { nfts, isLoading: isLoadingNFTs } = useFetchNFTs(
     nftWallets.map((wallet: Wallet) => wallet.id),
   );
 
+  const [progressBarWidth, setProgressBarWidth] = React.useState(-1);
+  const [validated, setValidated] = React.useState(0);
+  const nftRef = React.useRef(null);
+
+  useEffect(() => {
+    validateSha256Remote(false); // false parameter means only validate files smaller than MAX_FILE_SIZE
+    ipcRenderer.on('sha256DownloadProgress', progressListener);
+    ipcRenderer.on('sha256hash', gotHash);
+    return () => {
+      ipcRenderer.off('sha256DownloadProgress', progressListener);
+      ipcRenderer.off('sha256hash', gotHash);
+    };
+  }, []);
+
   const nft: NFTInfo | undefined = useMemo(() => {
     if (!nfts) {
       return;
     }
     return nfts.find((nft: NFTInfo) => nft.$nftId === nftId);
   }, [nfts]);
 
-  const { metadata, isLoading: isLoadingMetadata } = useNFTMetadata(nft);
+  nftRef.current = nft;
+
+  function progressListener(_event, progressObject: any) {
+    const nft = nftRef.current;
+    if (
+      nft &&
+      nft.dataUris &&
+      Array.isArray(nft.dataUris) &&
+      nft.dataUris[0] === progressObject.uri
+    ) {
+      setProgressBarWidth(progressObject.progress);
+      if (progressObject.progress === 1) {
+        setProgressBarWidth(-1);
+      }
+    }
+  }
+
+  function gotHash(_event, hash) {
+    if (nftRef.current) {
+      if (`0x${hash}` === nftRef.current.dataHash) {
+        setValidated(1);
+      } else {
+        setValidated(-1);
+      }
+    }
+  }
+
+  const { metadata, isLoading: isLoadingMetadata, error } = useNFTMetadata(nft);
+
+  const ValidateContainer = styled.div`
+    padding-top: 25px;
+    text-align: center;
+  `;
+
+  const ErrorMessage = styled.div`
+    color: red;
+  `;
 
   const isLoading = isLoadingWallets || isLoadingNFTs || isLoadingMetadata;
 
   if (isLoading) {
     return <Loading center />;
   }
 
-  function handleShowFullScreen() {
-    openDialog(<NFTPreviewDialog nft={nft} />);
```

Comment:
We want to keep this behavior for image NFTs. Clicking in the image area on the detail screen should show an enlarged version of the image.

Reply:
You'll want to set `pointer-events: none;` in the SandboxedIFrame component for image NFTs.

Reply:
Applied your patch.